### ISSUE TITLE
Skype 8 UWP: NVDA freezes or restarts when selecting text

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#9572]
### Steps to reproduce:
1. Open Skype UWP
2. Open a chat
3. In the message edit field, type "hello"
4. Press ctrl+home to go at the beginning of the message
5. Press ctrl+shift+right arrow key to select word by word and hold it down for more than 5 seconds

### Actual behavior:
NVDA will freeze or will restart with following in the log:
```
IO - inputCore.InputManager.executeGesture (23:46:36.671):
Input: kb(laptop):control+shift+rightArrow
IO - inputCore.logTimeSinceInput (23:46:36.701):
0.030 sec since input
IO - speech.speak (23:46:36.701):
Speaking [u'Hello ausgew\xe4hlt']
IO - inputCore.InputManager.executeGesture (23:46:36.861):
Input: kb(laptop):control+shift+rightArrow
IO - inputCore.InputManager.executeGesture (23:46:37.023):
Input: kb(laptop):control+shift+rightArrow
IO - inputCore.InputManager.executeGesture (23:46:37.512):
Input: kb(laptop):control+shift+rightArrow
IO - inputCore.InputManager.executeGesture (23:46:37.546):
Input: kb(laptop):control+shift+rightArrow
IO - inputCore.InputManager.executeGesture (23:46:37.578):
Input: kb(laptop):control+shift+rightArrow
IO - inputCore.InputManager.executeGesture (23:46:37.607):
Input: kb(laptop):control+shift+rightArrow
IO - inputCore.InputManager.executeGesture (23:46:37.957):
Input: kb(laptop):control+shift+rightArrow
DEBUGWARNING - watchdog._watcher (23:46:45.035):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 231, in <module>
  File "core.pyc", line 519, in main
  File "wx\core.pyc", line 2134, in MainLoop
  File "gui\__init__.pyc", line 1003, in Notify
  File "core.pyc", line 490, in run
  File "queueHandler.pyc", line 86, in pumpAll
  File "queueHandler.pyc", line 53, in flushQueue
  File "scriptHandler.pyc", line 150, in _queueScriptCallback
  File "scriptHandler.pyc", line 192, in executeScript
  File "editableText.pyc", line 329, in script_caret_changeSelection
  File "keyboardHandler.pyc", line 507, in send
  File "contextlib.pyc", line 17, in __enter__
  File "keyboardHandler.pyc", line 70, in ignoreInjection
```

### Expected behavior:
NVDA should not freeze or crash.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-17240,c20a5032

#### Windows version:
Win10 1809 Update

#### Name and version of other software in use when reproducing the issue:
Skype UWP 8.44.0.40

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2019.1.1 has the same issue. In Skype Electron app, the issue is not occuring.

**Comment by Adriani90** [comment]
cc: @josephsl, @ruifontes, @zstanecic, @k-kolev1985 could you please test and report your results? I am not sure if it is an issue only on my side. Derek also confirmed that NVDA freezes. In my case NVDA crashes and the process terminates automatically.

**Comment by ruifontes** [comment]
Using last Alpha, Windows 10 1809, compilation 17763, Skype Version 
8.44.0.40, application version 14.44.40.0 I don' have that problem...

Rui Fontes


Às 22:59 de 15/05/2019, Adriani90 escreveu:

**Comment by k-kolev1985** [comment]
Yes, I can definitely reproduce it. In my case NVDA froze and I had to restart it via its desktop shortcut.

### Test environment:
- Operating system: Windows 10 Pro version 1809 (build 17763.475), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: alpha-17248,7ffc65fe, installed, in bulgarian.
- Skype UWP Version: 8.45.76.40, app version: 14.45.40.0, in bulgarian.
- Processor: Intel Core i5-4670 at 3.40GHz.
- RAM Memory: 8.00GB.
- Graphics: MSI GeForce GTX-1050TI Gaming X 4G, 4096MB dedicated memory + 4065MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC887 at Intel Lynx Point PCH - High Definition Audio Controller.

I can provide a log, if necessary.

**Comment by ruifontes** [comment]
@Kolev, that version of Skype is the Skype Preview?
I don't find any way to update my version...

Rui Fontes


Às 07:51 de 16/05/2019, Kostadin Kolev escreveu:

**Comment by DrSooom** [comment]
@ruifontes: My Info » Settings » Help and Feedback: There you should see the current installed version of Skype 8 and if an update is available. Skype 8.45 was release two days ago – if I'm not wrong.

> 5. Press ctrl+shift+right arrow key to select word by word and hold it down for more than 5 seconds

All three keys or just CTRL+Shift? I wasn't able to reproduce this issue in Skype 8.45 on Win7x64 with NVDA 2018.1. Well, but now I know why I'm no longer using Skype on Win7. I just hate such Electron apps.

**Comment by k-kolev1985** [comment]
@ruifontes If you want to update to an Insider Preview of Skype UWP, you can join it from Skype's "About" dialog, if I remember correctly.

@DrSooom I think the problem described here affects only Skype UWP, which is the one from Microsoft Store for Windows 10.

**Comment by Adriani90** [comment]
This seems solved in Skype UWP 8.54 and NVDA 2019.3 Beta 1. Closing as works for me.
